### ISSUE TITLE
Unsupported opcode: RERAISE "# Decompile 'WITH_EXCEPT_START' is not implemented yet.

### ISSUE THREAD

**Issue by opticpp** [zrax/pycdc#317]
![image](https://user-images.githubusercontent.com/124639924/217116672-823cecb8-9dd6-4bb4-94c3-7aff87b29b06.png)
does anybody know how to fix that?

**Comment by TheHellTower** [comment]
> ![image](https://user-images.githubusercontent.com/124639924/217116672-823cecb8-9dd6-4bb4-94c3-7aff87b29b06.png) does anybody know how to fix that?

Hello, could you send a sample ? I seen in [3.9](https://docs.python.org/3.9/library/dis.html#opcode-WITH_EXCEPT_START) and [3.10](https://docs.python.org/3.10/library/dis.html#opcode-WITH_EXCEPT_START) it's the same and change a bit in [3.11](https://docs.python.org/3.11/library/dis.html#opcode-WITH_EXCEPT_START) so if you could send a not compiled script it would be helpful.

**Comment by Remi-Avec-Un-I** [comment]
```
import dis

def foo():
    try:
        with open('file.txt') as f:
            print(f.read())
    except Exception as e:
        print(f"Error: {e}")

dis.dis(foo)
``` For python 3.11, the result is:
```
...
        >>  128 PUSH_EXC_INFO
            130 WITH_EXCEPT_START
            132 POP_JUMP_FORWARD_IF_TRUE     4 (to 142)
...
``` Is that it?

**Comment by Faddi1** [comment]
Sir i need your help can you give me your WhatsApp number please? 🥺

**Comment by TheHellTower** [comment]
> ```
> import dis
> 
> def foo():
>     try:
>         with open('file.txt') as f:
>             print(f.read())
>     except Exception as e:
>         print(f"Error: {e}")
> 
> dis.dis(foo)
> ``` For python 3.11, the result is:
> ```
> 
> ... >> 128 PUSH_EXC_INFO 130 WITH_EXCEPT_START 132 POP_JUMP_FORWARD_IF_TRUE 4 (to 142) ...

I verified from 3.8 to 3.11 and:

- [`3.8`](https://docs.python.org/3.8/library/dis.html) need first: [`BEGIN_FINALLY`](https://docs.python.org/3.8/library/dis.html#opcode-BEGIN_FINALLY) implementation
- [`3.9`](https://docs.python.org/3.9/library/dis.html) need: [`your targeted opcode(WITH_EXCEPT_START)`](https://docs.python.org/3.9/library/dis.html#opcode-WITH_EXCEPT_START)
- [`3.10`](https://docs.python.org/3.10/library/dis.html) need: [`your targeted opcode(WITH_EXCEPT_START)`](https://docs.python.org/3.10/library/dis.html#opcode-WITH_EXCEPT_START)
- [`3.11`](https://docs.python.org/3.11/library/dis.html) need first: [`BEFORE_WITH`](https://docs.python.org/3.11/library/dis.html#opcode-BEFORE_WITH)

So when I will have time I will first work for 3.11 then 3.10-3.8.

**Comment by Faddi1** [comment]
Can i have your WhatsApp number?

**Comment by Faddi1** [comment]
![1679564666500](https://user-images.githubusercontent.com/101829391/227212808-d4b15711-bd52-44c2-aa7f-7d30fc5ca774.jpg)
Does anyone know how to decode this?

**Comment by tjengbudi** [comment]
> > ```
> > import dis
> > 
> > def foo():
> >     try:
> >         with open('file.txt') as f:
> >             print(f.read())
> >     except Exception as e:
> >         print(f"Error: {e}")
> > 
> > dis.dis(foo)
> > ``` For python 3.11, the result is:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ... >> 128 PUSH_EXC_INFO 130 WITH_EXCEPT_START 132 POP_JUMP_FORWARD_IF_TRUE 4 (to 142) ...
> 
> I verified from 3.8 to 3.11 and:
> 
> * [`3.8`](https://docs.python.org/3.8/library/dis.html) need first: [`BEGIN_FINALLY`](https://docs.python.org/3.8/library/dis.html#opcode-BEGIN_FINALLY) implementation
> * [`3.9`](https://docs.python.org/3.9/library/dis.html) need: [`your targeted opcode(WITH_EXCEPT_START)`](https://docs.python.org/3.9/library/dis.html#opcode-WITH_EXCEPT_START)
> * [`3.10`](https://docs.python.org/3.10/library/dis.html) need: [`your targeted opcode(WITH_EXCEPT_START)`](https://docs.python.org/3.10/library/dis.html#opcode-WITH_EXCEPT_START)
> * [`3.11`](https://docs.python.org/3.11/library/dis.html) need first: [`BEFORE_WITH`](https://docs.python.org/3.11/library/dis.html#opcode-BEFORE_WITH)
> 
> So when I will have time I will first work for 3.11 then 3.10-3.8.

Do you have fork repository for this? May i follow your progress?

**Comment by TheHellTower** [comment]
Hey guys,

giving a small "update" on the situation.

I didn't continue on pycdc because I moved in April to another country(`I'm from France originally`) and I don't really have much time for this project right now, I contributed to pycdc and I might do it again when I get a place and things like this but since I don't fully understand pycdc code I can't just throw my time on it.

In a few months I should have settled all that and should be able to continue contributing to it, sorry for the inconvenience !

Best Regards,
TheHellTower.
